collect_app/src/main/java/org/odk/collect/android/preferences/source/PreferencesDataStore.kt

```diff
@@ -0,0 +1,53 @@
+package org.odk.collect.android.preferences.source
+
+import androidx.preference.PreferenceDataStore
+
+class PreferencesDataStore(val preferencesDataSource: PreferencesDataSource) : PreferenceDataStore() {
```

Comment:
Ach my feedback has definitely pushed us towards some awkward naming here. I think this class should be named `<where data comes from>PreferenceDataStore` which would unfortunately mean calling this `PreferencesDataSourcePreferenceDataStore`. Yuck.

Maybe we should try and move away from using "preferences" for our own settings interfaces as Android seems to have done a good job of using every possible name using that word. Could we call our `PreferencesDataSource` something like `SettingsStore`, `ConfigStore`, `SettingsSource` etc?

Going to tag @lognaturel in as well because naming is hard.

Reply:
If class naming is the last thing maybe we could merge to avoid more conflicts and continue this discussion on slack?

Reply:
I have a preference for resolving the issue in this PR if possible to try to keep history somewhat easier to navigate. Big renames can be really hard to follow. What are you seeing conflicts with, @grzesiek2010? Always helpful to make conflict concerns concrete so it's easier to evaluate! Looks like some of the renaming needs to happen in master anyway so maybe it's not a big deal in this case.

I think using the word `Settings` for Collect-specific classes is a good idea since that matches the UI. Some ideas:
`PreferencesDataSource` -> `Settings`
`SharedPreferencesDataSource` -> `SharedPreferencesSettings`
`PreferencesDataSourceProvider` -> `SettingsProvider` (`getMetaSettings`, `getGeneralSettings`, etc)

Reply:
Yeah for me, it's easier to make the changes here as I'm going to have to rebase on top either way. I'm happy with the general change to "Settings".

Reply:
Ok renamed those classes/methods.